src/CMakeLists.txt

```diff
@@ -69,17 +69,38 @@ if ("${JUPYTER}" STREQUAL "ON")
    add_definitions(-DRB_XCODE)
 endif()
 
+#MESSAGE("Boost location hints:")
+#if (BOOST_ROOT)
+#  MESSAGE("  BOOST_ROOT: ${BOOST_ROOT}")
+#else()
+#  MESSAGE("  BOOST_ROOT not set")
+#endif()
+#if (BOOST_INCLUDEDIR)
+#  MESSAGE("  BOOST_INCLUDEDIR: ${BOOST_INCLUDEDIR}")
+#endif()
+#if (BOOST_LIBRARYDIR)
+#  MESSAGE("  BOOST_LIBRARYDIR: ${BOOST_LIBRARYDIR}")
+#endif()
 MESSAGE("Boost location hints:")
+if (BOOST_ROOT OR BOOST_INCLUDEDIR OR BOOST_LIBRARYDIR) 
+  set(Boost_NO_SYSTEM_PATHS ON)
+  set(Boost_NO_BOOST_CMAKE ON)
+  set(Boost_NO_WARN_NEW_VERSION ON)
+endif()
 if (BOOST_ROOT)
   MESSAGE("  BOOST_ROOT: ${BOOST_ROOT}")
-else()
-  MESSAGE("  BOOST_ROOT not set")
+  set(Boost_INCLUDE_DIR ${BOOST_ROOT}/include)
+  set(Boost_LIBRARY_DIR ${BOOST_ROOT}/lib)
```

Comment:
Similarly for this -- I'm not sure we should do this.
Also, I think that `${BOOST_ROOT}/lib` isn't always the right location for the libraries.

Reply:
Same as above